### ISSUE TITLE
[web-animations-1] 4.3.2. Document timelines, logic

### ISSUE THREAD

**Issue by verbessern** [w3c/csswg-drafts#5420]
The phrase "If the associated document is not an active document or its time origin has not yet been established", makes no sense to me, because the time origin, is defined in HR time in "4.2 time Origin attribute" as `DOMHighResTimeStamp`, that is defined to be in WebIDL `typedef double DOMHighResTimeStamp` and double itself is defined in the WebIDL in "2.13.14. double" as `finite`. What I mean is that time origin, is always established (never unresolved, because its not optional, neither its an unrestricted double), so the text after "active document" is technically never going to lead to 'true'. 

Also the first paragraph of this chapter is foggy somehow. In a common sense, it looks like that the document start time (`now` at the start) is subtracted from `now` at update time to have the `elapsed time`. Then `origin-relative time = elapsed time + origin offset time`. That means that for a non-default time line with `origin offset time = 10s`, at the start of the animation, the `current time = 10s`, because its a sum, if it was a subtraction, then it was going to be `-10s`, because the origin is actually at the 10s. Is it really a plus intended?

**Comment by birtles** [comment]
Yes, you're right. The intention is that there would be, in future, a subclass of document timeline whose time origin would be relevant to some event. This is needed for SVG animations where animations are relative to timing of the onload event of the corresponding outermost document fragment.

As for the subtraction vs addition issue, again, I believe you are correct. (Note to self to check Gecko's implementation of this, however.)

**Comment by verbessern** [comment]
I think you mean that there will be an extension to the timeline, not the document timeline. That expression probably will better fit to this new class and is really not relevant to the document timeline itself, but its a draft, so...i will close the issue.

**Comment by birtles** [comment]
The intention is to extend document timeline itself by having an origin time that is initially unresolved and presumably updating the IDL to allow that.

**Comment by birtles** [comment]
> As for the subtraction vs addition issue, again, I believe you are correct. (Note to self to check Gecko's implementation of this, however.)

I had a look at what Gecko does, and [it appears to add the origin time](https://searchfox.org/mozilla-central/rev/2f9eacd9d3d995c937b4251a5557d95d494c9be1/dom/animation/DocumentTimeline.cpp#264-265). I tried to create a test that proves this is wrong but couldn't ([test here](https://codepen.io/birtles/pen/jOqrPdX)) so I need to try again to nut this out.

**Comment by verbessern** [comment]
You are right. I've got confused by the paragraphs in the chapter, but after I'm reading it multiple times, I understand it better. The phrase `whose current time is calculated as a fixed offset from the now timestamp` actually is that minus I was talking about, and the last paragraph is the plus (the reverse function) that you have found. The text is somehow complex, but after I've got familiar with the terminology, its readable. Also this 'origin' relative/time/offsets added to the mix. I think its all solved now.

**Comment by verbessern** [comment]
The think is that `origin-relative time = document start time + timeline time` not as I previously thought.
